arcgis-ios-sdk-samples/Analysis/Line of sight (geoelement)/LineOfSightGeoElementViewController.swift

```diff
@@ -0,0 +1,204 @@
+        AGSPoint(x: -73.984513, y: 40.748469, spatialReference: AGSSpatialReference.wgs84())
+    ]
+
+    // handle onto any line of sight KVO observer
+    private var losObserver:NSKeyValueObservation?
+
+    private var initialViewpointCenter:AGSPoint {
+        // If possible, find the middle of the block that the taxi will drive around, or else focus on the observer
+        return AGSGeometryEngine.unionGeometries(streetIntersectionLocations)?.extent.center ?? observerPoint
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        // initialize the scene with an imagery basemap
+        scene = AGSScene(basemap: AGSBasemap.imageryWithLabels())
+
+        // initialize the elevation source and add it to the base surface of the scene
+        let elevationSrc = AGSArcGISTiledElevationSource(url: .worldElevationService)
+        scene.baseSurface?.elevationSources.append(elevationSrc)
+
+        // add some buildings to the scene
+        let sceneLayer = AGSArcGISSceneLayer(url: .newYorkBuildingsService)
+        scene.operationalLayers.add(sceneLayer)
+
+        // initialize the taxi graphic
+        let taxiSymbol = AGSModelSceneSymbol(name: "dolmus", extension: "3ds", scale: 1)
+        taxiSymbol.anchorPosition = .bottom
+        taxiGraphic = AGSGraphic(geometry: streetIntersectionLocations[0], symbol: taxiSymbol, attributes: nil)
+
+        // initialize the observer graphic
+        let observerSymbol = AGSSimpleMarkerSceneSymbol(style: .sphere, color: .red, height: 10, width: 10, depth: 10, anchorPosition: .center)
+        observerGraphic = AGSGraphic(geometry: observerPoint, symbol: observerSymbol, attributes: nil)
+
+        // initialize the graphics overlay
+        overlay = AGSGraphicsOverlay()
+        overlay.sceneProperties = AGSLayerSceneProperties(surfacePlacement: .relative)
+
+        // add the taxi and observer to the graphics overlay
+        overlay.graphics.addObjects(from: [observerGraphic, taxiGraphic])
+
+        super.init(coder: aDecoder)
+
+        // set the initial viewpoint
+        scene.initialViewpoint = AGSViewpoint(center: initialViewpointCenter, scale: 6000)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["LineOfSightGeoElementViewController"]
+
+        // assign the scene to the scene view
+        sceneView.scene = scene
+
+        // Add a graphics overlay for the taxi and observer
+        sceneView.graphicsOverlays.add(overlay)
+
+        // initialize the line of sight analysis
+        lineOfSight = AGSGeoElementLineOfSight(observerGeoElement: observerGraphic, targetGeoElement: taxiGraphic)
+
+        // update the UI if the Line of Sight analysis result changes
+        losObserver = lineOfSight?.observe(\.targetVisibility, options: [.new], changeHandler: { (losAnalysis, change) in
```

Comment:
You're welcome. The reason is because the type of the parameter is `NSKeyValueObservingOptions` and the type of of `new` is the same. As you know, including the brackets turn it into a literal array of KVO options. Since the type conforms to `OptionSet` which conforms to `ExpressibleByArrayLiteral`, a new KVO options instance is created from the array rather than just using the statically defined value. Does that make sense?